coderd/templates.go

```diff
@@ -101,11 +106,37 @@ func (api *API) deleteTemplate(rw http.ResponseWriter, r *http.Request) {
 		})
 		return
 	}
+
+	api.notifyTemplateDeleted(ctx, template, apiKey.UserID)
+
 	httpapi.Write(ctx, rw, http.StatusOK, codersdk.Response{
 		Message: "Template has been deleted!",
 	})
 }
 
+func (api *API) notifyTemplateDeleted(ctx context.Context, template database.Template, initiatorID uuid.UUID) {
+	if template.CreatedBy == initiatorID {
+		return
+	}
```

Comment:
We should be enumerating all the template admins and owners here and enqueueing one notification each.
If an admins is the initiator, you can suppress that notification.

Take a look at `coderd/users.go` where `TemplateUserAccountCreated` is triggered.

Reply:
I was wondering about this yesterday 🤔